network/src/peer/test.rs

```diff
@@ -191,8 +191,6 @@ fn peer_send_message() {
 
     let client = async move {
         peer_handle.send_message(send_msg, PROTOCOL).await.unwrap();
-        //ManuallyDrop::new(peer_handle);
-        forget(peer_handle);
```

Comment:
@phlip9 I suspect this is safe to remove, neither the ManuallyDrop nor the forget were needed for the test to pass.   Don't know if this introduces any kind of race though?

Reply:
Yeah just change this to

```rust
     let (
         peer,
         peer_handle,
         _connection,
         _peer_notifs_rx,
         _peer_rpc_notifs_rx,
         _peer_direct_send_notifs_rx,
     ) = build_test_peer(rt.handle().clone(), ConnectionOrigin::Inbound);
     // Perform yamux handshake in a different task.
     let drop = async move {
         // Drop peer handle.
         drop(peer_handle);
     };
```

Reply:
Oh that's what you already did, oops 😅 

The original intention was to make sure the peer actor shutdown correctly if you dropped the peer_handle (but not the underlying connection). Of course, we can actually accomplish that without the `ManuallyDrop` by just changing `connection` to `_connection` so it drops at the end of the test : )